src/Native/System.Native/pal_networking.cpp

```diff
@@ -1197,8 +1201,12 @@ extern "C" Error SystemNative_GetIPv4MulticastOption(int32_t socket, int32_t mul
     }
 
     *option = {.MulticastAddress = opt.imr_multiaddr.s_addr,
+#if HAVE_IP_MREQN
                .LocalAddress = opt.imr_address.s_addr,
                .InterfaceIndex = opt.imr_ifindex};
+#else
+               .LocalAddress = opt.imr_interface.s_addr};
```

Comment:
IMO we should probably still set `InterfaceIndex` explicitly (AFAIK 0 would be okay).